### PR TITLE
ts 5.6 and other updates

### PR DESCRIPTION
**Pull request by jnsdls**
## PR-Codex overview
The focus of this PR is updating TypeScript and package versions across multiple packages.

### Detailed summary
- Updated `typescript` to version `5.6.2` in various package.json files
- Updated `tsup` to version `8.2.4`
- Added `usePagination` in multiple files
- Fixed code issues like handling `store.delete` and `useTotal` passing
- Updated dependencies like `next`, `posthog-js`, `lucide-react`, and more

> The following files were skipped due to too many changes: `packages/thirdweb/src/wallets/in-app/core/eip5972/in-app-wallet-calls.ts`, `legacy_packages/wallets/src/evm/connectors/gcp-kms/utils/gcp-kms-utils.ts`, `apps/dashboard/package.json`, `packages/thirdweb/package.json`, `legacy_packages/contracts-js/package.json`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
* **#4522** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4522?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jnsdls and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/contract-ui/tabs/analytics/page.tsx [hunk 1]
```diff
@@ -321,6 +321,8 @@ const AnalyticsData: React.FC<AnalyticsStatProps> = ({
   contractAddress,
   useTotal,
 }) => {
+  // FIXME: re-work this to not pass the hook down
+  // eslint-disable-next-line react-compiler/react-compiler
```
Comment (joaquim-verges) [inline apps/dashboard/src/contract-ui/tabs/analytics/page.tsx:325]: whats the issue here?

File: packages/thirdweb/package.json [hunk 2]
```diff
@@ -307,11 +271,11 @@
     "build:generate": "bun scripts/generate/generate.ts",
     "build:generate-wallets": "bun scripts/wallets/generate.ts",
     "dev": "tsc --project ./tsconfig.build.json --module es2020 --outDir ./dist/esm --watch",
-    "dev:cjs": "printf '{\"type\":\"commonjs\"}' > ./dist/cjs/package.json && tsc --project ./tsconfig.build.json --module commonjs --outDir ./dist/cjs --verbatimModuleSyntax false --watch",
-    "dev:esm": "printf '{\"type\": \"module\",\"sideEffects\":false}' > ./dist/esm/package.json && tsc --project ./tsconfig.build.json --module es2020 --outDir ./dist/esm --watch",
+    "dev:cjs": "printf '{\"type\":\"commonjs\"}' > ./dist/cjs/package.json && tsc --noCheck --project ./tsconfig.build.json --module commonjs --outDir ./dist/cjs --verbatimModuleSyntax false --watch",
+    "dev:esm": "printf '{\"type\": \"module\",\"sideEffects\":false}' > ./dist/esm/package.json && tsc --noCheck --project ./tsconfig.build.json --module es2020 --outDir ./dist/esm --watch",
```
Comment (joaquim-verges) [inline packages/thirdweb/package.json:275]: noCheck? seems scary

File: packages/thirdweb/src/chains/chain-definitions/polygon.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { defineChain } from "../utils.js";
 export const polygon = /*@__PURE__*/ defineChain({
   id: 137,
   name: "Polygon",
-  nativeCurrency: { name: "MATIC", symbol: "MATIC", decimals: 18 },
+  nativeCurrency: { name: "POL", symbol: "POL", decimals: 18 },
```
Comment (joaquim-verges) [inline packages/thirdweb/src/chains/chain-definitions/polygon.ts:9]: sneaking it in huh xD

Comment (joaquim-verges) [reply packages/thirdweb/src/chains/chain-definitions/polygon.ts:9]: this prob needs a changeset on its own, would separate it tbh

File: packages/thirdweb/package.json (hunk 2 above)
Comment (jnsdls) [reply packages/thirdweb/package.json:275]: read more: https://devblogs.microsoft.com/typescript/announcing-typescript-5-6/#the---nocheck-option6

File: packages/thirdweb/src/chains/chain-definitions/polygon.ts (hunk 3 above)
Comment (jnsdls) [reply packages/thirdweb/src/chains/chain-definitions/polygon.ts:9]: eh its already changed on chain api I just forgot it here

File: apps/dashboard/src/contract-ui/tabs/analytics/page.tsx (hunk 1 above)
Comment (joaquim-verges) [reply apps/dashboard/src/contract-ui/tabs/analytics/page.tsx:325]: oh i missed it in the props, yeah thats weird.

File: packages/thirdweb/package.json (hunk 2 above)
Comment (joaquim-verges) [reply packages/thirdweb/package.json:275]: i see

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**